nncf/__init__.py

```diff
@@ -22,28 +23,39 @@
 from nncf.quantization import quantize
 from nncf.quantization import quantize_with_accuracy_control
 
+_LOADED_FRAMEWORKS = {
+    "torch": True,
+    "tensorflow": True,
+    "onnx": True,
+    "openvino": True
+}
+
 try:
     import torch
 except ImportError:
-    torch = None
+    _LOADED_FRAMEWORKS["torch"] = False
 
 try:
     import tensorflow as tf
 except ImportError:
-    tf = None
+    _LOADED_FRAMEWORKS["tensorflow"] = False
 
 try:
     import onnx
 except ImportError:
-    onnx = None
+    _LOADED_FRAMEWORKS["onnx"] = False
 
 try:
     import openvino.runtime as ov_runtime
     import openvino.tools.pot as ov_pot
 except ImportError:
-    ov_runtime = None
-    ov_pot = None
+    _LOADED_FRAMEWORKS["openvino"] = False
 
-if torch is None and tf is None and onnx is None and ov_runtime is None and ov_pot is None:
-    import warnings
-    warnings.warn("None of PyTorch, TensorFlow, ONNX, OpenVINO have been found. Please, install one of the frameworks")
+from nncf.common.logging import nncf_logger
+if not any(_LOADED_FRAMEWORKS.values()):
+    nncf_logger.error("Neither PyTorch, TensorFlow, ONNX or OpenVINO Python packages have been found in your Python "
+                      "environment.\n"
+                      "Please install one of the supported frameworks above in order to use NNCF on top of it.")
```

Comment:
Since you get a warning when you use an unsupported version of PyTorch or Tensorflow, would it be good to link to the installation guide that mentions this?

Reply:
Done